src/nanotron/metrics_logging.py

```diff
@@ -43,36 +42,23 @@ def get_attribute_by_path(obj: Any, path: str, debug: bool = False) -> Optional[
 ########################
 
 
-def compute_entropy(tensor: torch.Tensor) -> torch.Tensor:
-    """Compute Shannon entropy of a tensor."""
-    if tensor.min() < 0 or tensor.sum() != 1.0:
-        if tensor.min() < 0:
-            probs = F.softmax(tensor.float(), dim=-1)
-        else:
-            probs = tensor.float() / tensor.sum()
-    else:
-        probs = tensor.float()
-    # Compute entropy: -sum(p * log(p))
-    # Add epsilon to avoid log(0)
-    entropy = -torch.sum(probs * torch.log2(probs + 1e-10))
-    return entropy
-
-
 def compute_kurtosis(tensor: torch.Tensor) -> torch.Tensor:
     """
     Compute kurtosis which measures "tailedness" of distribution.
     High kurtosis indicates outliers, low kurtosis indicates uniform-like distribution.
     """
-    tensor = tensor.flatten().float()
-    mean = tensor.mean()
-    std = tensor.std()
+    # Use float() directly on tensor operations rather than making a copy
+    mean = tensor.float().mean()
+    std = tensor.float().std()
```

Comment:
you're doing tensor.float() too many separate times,
you can instead do 
`tensor = tensor.float()`
in a line above